Modules/mmapmodule.c

```diff
@@ -896,6 +902,54 @@ mmap_madvise_method(mmap_object *self, PyObject *args)
 }
 #endif // HAVE_MADVISE
 
+#if defined(MS_WINDOWS) || defined(HAVE_MPROTECT)
+/*[clinic input]
+mmap.mmap.mprotect
+    prot: int
+    start: Py_ssize_t = 0
+    length: Py_ssize_t = -1
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+mmap_mmap_mprotect_impl(mmap_object *self, int prot, Py_ssize_t start,
+                        Py_ssize_t length)
+/*[clinic end generated code: output=5d0923571db44207 input=5632169def6c8789]*/
+{
+    CHECK_VALID(NULL);
+
+    if (start < 0 || start >= self->size) {
+        PyErr_SetString(PyExc_ValueError, "mprotect start out of bounds");
+        return NULL;
+    }
+    if (length == -1) {
```

Comment:
What if the user passes the negative value?
Just simply..?
```suggestion
    if (length < 0) {
```